src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -817,6 +833,13 @@ public GraphBuildSubmission PendBuildRequest(GraphBuildRequestData requestData)
                 VerifyStateInternal(BuildManagerState.Building);
 
                 var newSubmission = new GraphBuildSubmission(this, GetNextSubmissionId(), requestData);
+
+                if (KnownTelemetry.BuildTelemetry != null)
+                {
+                    KnownTelemetry.BuildTelemetry.Project ??= requestData.ProjectGraphEntryPoints?.FirstOrDefault().ProjectFile;
```

Comment:
Can there be multiple graph entry points? If so, is this right?

Reply:
There can be, however, this is probably very rare situation and we are fine to have it slightly wrong.
'BuildTelemetry.Project' is designed to be used for events correlations it will correlate build telemetry even when it include just one entry point.